rackcity/api/serializers/asset_serializers.py

```diff
@@ -110,24 +116,74 @@ class BulkAssetSerializer(serializers.ModelSerializer):
     )
     # by default, calls get_<field> - in this case, get_rack
     rack = serializers.SerializerMethodField()
-    rack_position = serializers.IntegerField(source='rack_position')
+    power_port_connection_1 = serializers.SerializerMethodField()
+    power_port_connection_2 = serializers.SerializerMethodField()
 
     class Meta:
         model = Asset
         fields = (
             'asset_number',
             'hostname',
+            'datacenter',
             'rack',
             'rack_position',
             'vendor',
             'model_number',
             'owner',
-            'comment'
+            'comment',
+            'power_port_connection_1',
+            'power_port_connection_2'
         )
 
     def get_rack(self, asset):
         return asset.rack.row_letter + str(asset.rack.rack_num)
 
+    def get_power_port_connection_1(self, asset):
+        return self.power_port_connection(asset, port_number=1)
+
+    def get_power_port_connection_2(self, asset):
+        return self.power_port_connection(asset, port_number=2)
+
+    def power_port_connection(self, asset, port_number):
+        power_ports = PowerPort.objects.filter(asset=asset.id)
+        if (
+            not power_ports
+            or len(power_ports) < port_number
+        ):
+            return None
+        power_port = power_ports.get(port_name=str(port_number))
+        if (
+            not power_port.power_connection
+        ):
+            return None
+        pdu_port = power_port.power_connection
+        if (
+            not pdu_port.left_right
+            or not pdu_port.port_number
+        ):
+            return None
+        return pdu_port.left_right+str(pdu_port.port_number)
+
+
+def normalize_bulk_asset_data(bulk_asset_data):
+    power_connections = {}
+    if bulk_asset_data['power_port_connection_1']:
+        power_connections["1"] = {
+            "left_right": bulk_asset_data['power_port_connection_1'][:1],
+            "port_number": int(bulk_asset_data['power_port_connection_1'][1:])
+        }
+    if bulk_asset_data['power_port_connection_2']:
+        power_connections["2"] = {
+            "left_right": bulk_asset_data['power_port_connection_2'][:1],
+            "port_number": int(bulk_asset_data['power_port_connection_2'][1:])
+        }
+    bulk_asset_data['power_connections'] = power_connections
+    del bulk_asset_data['power_port_connection_1']
+    del bulk_asset_data['power_port_connection_2']
```

Comment:
need to ensure we never get a KeyError on del. are we just assuming the bulk import file will be formatted correctly, will all specified fields included but some might be null?

Reply:
JK I see that you validate all column names in assets views, so that answers my question